apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -348,6 +346,47 @@ const DanceAiModal: React.FunctionComponent = () => {
         : aiBotYes;
   }
 
+  const headerValue = () => {
+    return (
+      <div
+        className={moduleStyles.inputsContainer}
+        onClick={handleStartOverClick}
+      >
+        {Array.from(Array(SLOT_COUNT).keys()).map(index => {
+          const item = getItem(inputs[index]);
+          return (
+            <div key={index} className={moduleStyles.emojiSlot}>
+              {item && (
+                <EmojiIcon item={item} className={moduleStyles.emojiSlotIcon} />
+              )}
+            </div>
+          );
+        })}
+      </div>
+    );
+  };
+
+  // The header comes from a localized string like this: "generate {input} effect".
+  // We split the localized string on the "{input}", and rebuild the HTML but with
+  // the emoji box in place of that "{input}".
+  const INPUT_KEY = '__CDO_INPUTS__';
+  const headerTextSplit = i18n
+    .danceAiModalHeading({input: INPUT_KEY})
+    .split(INPUT_KEY);
+  const headerContent = [0, 1, 2]
+    .map(index => {
+      if (index === 0) {
+        return headerTextSplit[0];
+      } else if (index === 1) {
+        return headerValue();
+      } else {
+        return headerTextSplit[1];
+      }
+    })
+    .map((part: string, index: number) => {
```

Comment:
Should we filter out empty strings before mapping here? If the translation is something like `{input} generate effects` we'd have an empty string at the beginning right?

Reply:
Good question.  I'm actually counting on those potentially empty strings being in the three-element array when referencing by index just below.